bundles/org.openhab.core.config.discovery.usbserial.otheros/src/main/java/org/openhab/core/config/discovery/usbserial/otheros/internal/UsbProductDatabase.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.config.discovery.usbserial.otheros.internal;
+
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * This is a database map between hex 'vendorId:productId' pairs and the descriptions of known products. This is
+ * required because in some cases the USB scanners fail to read the product description strings, so this provides
+ * fallback descriptions.
+ * <p>
+ * It contains a map of vendor ids, product ids, and product descriptions. The descriptions should include at least a
+ * tag indicating the type of binding it supports. The entries MUST be formatted as follows:
+ * <p>
+ * <li>keys e.g. "vendorId:productId" -- each part comprising four <b>UPPER</b>-case hex characters.
+ * <li>entries e.g. "vendor OEM / sub-vendor name / ZigBee or Z-Wave" -- including a supported binding id.
+ * <p>
+ * Note: supported binding ids can be as follows:
+ * <li>'Zigbee'
+ * <li>'Z-Wave'
+ * <li>'EnOcean'
+ * Please do NOT include supported binding id in the case of very commonly used generic serial chips, since this may
+ * lead to false-positive discoveries.
+ *
+ * @author Andrew Fiddian-Green - Initial contribution
+ */
+@NonNullByDefault
+public class UsbProductDatabase {
+
+    // @formatter:off
+    public static final Map<String, String> PRODUCT_NAMES = Map.of(
```

Comment:
Wouldn't it be better to move these definitions into a separate resource file rather than embedding them in Java?

Reply:
IDK. A file is a file. Whether or not its name ends in .java or .json or .xml or .txt .. Or?? The advantage of a .java file is that a) it can be documented via javadoc, and b) requires no extra overhead to parse. Ultimately the data is consumed in the java finder anyway..

Reply:
Well, based on that argument, we can get rid of all the XML files that describe things, bindings, channels etc, and just code everything into the java source?

The advantage of separating it out (IMHO) is maintainability. Having a defined file that allows non-coders to add data without getting into Java is (again, IMHO) better and cleaner. Yes, it requires some work to parse a new file, but that's life - we could have saved loads of work on OH core by just coding all definitions directly in the source...

Reply:
Ok.

Reply:
=> continue discussion in #3930